toolbox/py_xnvme_pkg/Makefile

```diff
@@ -0,0 +1,123 @@
+#
+# This Makefile serves as convenient command-line auto-completion when working on xNVMe-ctypes
+#
+#
+TOOLBOX_DIR?=../
+PROJECT_NAME=xnvme
+XNVME_CAPI_HEADERS=$(shell ls ../../include/libxnvme* | grep -v util | grep -v libxnvmec)
+XNVME_SO_PATH=$(shell pkg-config xnvme --variable=libdir)/libxnvme-shared.so
+
+define default-help
+# invoke: 'make uninstall', 'make install'
+endef
+.PHONY: default
+default: uninstall install
+	@echo "## py: make default"
+	@echo "## py: make default [DONE]"
+
+define build-help
+# invoke: 'python3 setup.py build'
+endef
+.PHONY: build
+build:
+	@echo "## py: make build"
+	@python3 setup.py build
+	@echo "## py: make build [DONE]"
+
+define build-sdist-help
+# Create a source package (invoke: 'python3 setup.py sdist')
+endef
+.PHONY: build-sdist
+build-sdist:
+	@echo "## py: make build-sdist"
+	@python3 setup.py sdist
+	@echo "## py: make build-sdist [DONE]"
+
+define install-help
+# install for current user
+endef
+.PHONY: install
+install:
+	@echo "## py: make install"
+	@pip3 install . --user
+	@echo "## py: make install [DONE]"
+
+define uninstall-help
+# uninstall
+#
+# Prefix with 'sudo' when uninstalling a system-wide installation
+endef
+.PHONY: uninstall
+uninstall:
+	@echo "## py: make uninstall"
+	@pip3 uninstall ${PROJECT_NAME} --yes || echo "Cannot uninstall => That is OK"
+	@echo "## py: make uninstall [DONE]"
+
+define install-system-help
+# install system-wide
+#
+# install system-wide
+endef
+.PHONY: install-system
+install-system:
+	@echo "## py: make install-system"
+	@pip3 install .
+	@echo "## py: make install-system [DONE]"
+
+define test-help
+# Run pytest on tests/
+endef
+.PHONY: test
+test:
+	@echo "## py: make test"
+	@pytest tests
+	@echo "## py: make test [DONE]"
+
+define generate-ctypes-help
+# Generate definitions (xnvme_ctypes.header) from xNVMe public C API headers
+endef
+.PHONY: generate-ctypes
+generate-ctypes:
+	@echo "## py: gen"
+	@clang2py ${XNVME_CAPI_HEADERS} > xnvme/ctypes_mapping.py
+	@echo "## Remember to also call 'make format-all' to fix format"
```

Comment:
`format-all ` should probably be `patch-ctypes`

Reply:
Right! Thanks!